### PR TITLE
Pulsar Connector Improvements/Fixes

### PR DESCRIPTION
**Pull request by ufukyilmaz**
### What this PR does / why do we need it:
These changes address most of Ondrej's reviews.
In this PR:
1. Pulsar source creation APIs is changed to use builder.
2. The issue that not required fields exist in the sink builder's entry point is solved.
3. Update JavaDocs, README, and tests according to changes.

#### Checklist

- [x] Signed the Hazelcast CLA
- [x] No checkstyle issues (`./gradlew check`)
- [x] No tests failures (`./gradlew test`)
- [x] Documentation which complies with README template (see `templates/README.template.md`).

### CODE REVIEW

File: pulsar/src/main/java/com/hazelcast/jet/contrib/pulsar/PulsarReaderBuilder.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    private final SupplierEx<PulsarClient> connectionSupplier;
+    private Map<String, Object> readerConfig;
+    private SupplierEx<Schema<M>> schemaSupplier;
+    private FunctionEx<Message<M>, T> projectionFn;
+
+
+    /**
+     * Required fields of Pulsar reader
+     *
+     * @param topic              Pulsar topic name to publish to
```
[the remark this answers is not in the document]
Comment (ufukyilmaz) [reply pulsar/src/main/java/com/hazelcast/jet/contrib/pulsar/PulsarReaderBuilder.java]: fixed it

File: pulsar/src/main/java/com/hazelcast/jet/contrib/pulsar/PulsarSources.java [hunk 2]
```diff
@@ -82,283 +65,138 @@ private PulsarSources() {
...
-                .destroyFn(ConsumerContext::destroy)
-                .distributed(preferredLocalParallelism)
-                .build();
+        return consumerSrcBuilder(Collections.singletonList(topic), connectionSupplier, schemaSupplier, projectionFn);
     }
 
 
     /**
-     * Creates a fault-tolerant timestamped {@link StreamSource}
+     * Convenience for {@link #readerSrcBuilder(String, SupplierEx, SupplierEx, FunctionEx)}.
```
[the remark this answers is not in the document]
Comment (ufukyilmaz) [reply pulsar/src/main/java/com/hazelcast/jet/contrib/pulsar/PulsarSources.java]: fixed it.

File: pulsar/src/main/java/com/hazelcast/jet/contrib/pulsar/PulsarSources.java [hunk 3]
```diff
@@ -82,283 +65,138 @@ private PulsarSources() {
...
+     * <p>
+     *
+     * @param topic              the single topic to consume
+     * @param connectionSupplier Pulsar client supplier
+     * @param schemaSupplier     supplies the schema for consuming messages
+     * @param projectionFn       converts a Pulsar message to an emitted item.
+     * @param <M>                the type of the message read by {@code pulsarReader}
+     * @param <T>                the type of data emitted from {@code StreamSource}
+     */
+    public static <M, T> StreamSource<T> pulsarConsumer(
```
[the remark this answers is not in the document]
Comment (ufukyilmaz) [reply pulsar/src/main/java/com/hazelcast/jet/contrib/pulsar/PulsarSources.java:131]: I renamed them

File: pulsar/src/main/java/com/hazelcast/jet/contrib/pulsar/PulsarSources.java [hunk 4]
```diff
@@ -82,283 +65,138 @@ private PulsarSources() {
...
+    ) {
+        return new PulsarConsumerBuilder<>(topics, connectionSupplier, schemaSupplier, projectionFn);
+    }
+
+    /**
+     * See the {@link #consumerSrcBuilder(List, SupplierEx, SupplierEx, FunctionEx)}
+     * It gets a single String for the topic name in case it should read only
+     * from a single topic.
+     *
+     * @param topic the single topic to consume
```
[the remark this answers is not in the document]
Comment (ufukyilmaz) [reply pulsar/src/main/java/com/hazelcast/jet/contrib/pulsar/PulsarSources.java:99]: done

**Comment by hz-highfive** [comment]
The job `jet-contrib-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/jet-contrib-pr-builder/326/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/jet-contrib-pr-builder/326/jet-contrib-pr-builder-326.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>

</details>
</pre>

**Pull request merged**